sdk/inc/azure/core/az_log.h

```diff
@@ -64,56 +61,63 @@ enum az_log_classification_core
- * @param[in] classifications __[nullable]__ An array of #az_log_classification values, terminated
- * by #AZ_LOG_END_OF_LIST.
+ * @return Whether or not a log message with the provided classification should be logged.
+ */
+typedef bool (*az_log_should_write_fn)(az_log_classification classification);
+
+/**
+ * @brief Sets the functions that will be invoked to report an SDK log message.
  *
- * @details If no classifications are set (\p classifications is `NULL`), the application will
- * receive log messages for all #az_log_classification values.
- * @details If \p classifications is not `NULL`, it must point to an array of
- * #az_log_classification, terminated by #AZ_LOG_END_OF_LIST.
- * @details In contrast to \p classifications being `NULL`, \p classifications pointing to an empty
- * array (which still should be terminated by #AZ_LOG_END_OF_LIST), states that an application is
- * not interested in receiving any log messages.
+ * @param[in] az_log_message_callback __[nullable]__ A pointer to the function that will be invoked
+ * when the SDK reports a log message that should be logged according to the result of the
+ * #az_log_should_write_fn provided to #az_log_set_filter_callback(). If `NULL`, no function will be
+ * invoked.
  *
- * @warning Users must not change the \p classifications array elements, once it is passed to this
- * function. If \p classifications array is allocated on a stack, program behavior in multi-threaded
- * environment is undefined.
+ * @remarks By default, this is `NULL`, which means, no function is invoked.
  */
-#ifndef AZ_NO_LOGGING
-void az_log_set_classifications(az_log_classification const classifications[]);
-#else
-AZ_INLINE void az_log_set_classifications(az_log_classification const classifications[])
+#ifdef AZ_NO_LOGGING
+AZ_INLINE void az_log_set_message_callback(az_log_message_fn az_log_message_callback)
 {
-  (void)classifications;
+  (void)az_log_message_callback;
 }
+#else
+void az_log_set_message_callback(az_log_message_fn az_log_message_callback);
 #endif // AZ_NO_LOGGING
 
 /**
- * @brief Sets the function that will be invoked to report an SDK log message.
+ * @brief Sets the functions that will be invoked to check whether an SDK log message should be
+ * reported.
  *
- * @param[in] az_log_message_callback __[nullable]__ A pointer to the function that will be invoked
- * when the SDK reports a log message matching one of the #az_log_classification passed to
- * #az_log_set_classifications(). If `NULL`, no function will be invoked.
+ * @param[in] az_log_should_write_callback __[nullable]__ A pointer to the function that will be
+ * invoked when the SDK checks whether a log message of a particular #az_log_classification should
+ * be logged. If `NULL`, log messages for all classifications will be logged, by passing them to the
+ * #az_log_message_fn provided to #az_log_set_message_callback().
+ *
+ * @remarks By default, this is `NULL`, in which case no function is invoked to check whether a
```

Comment:
We didn't say that on `set_classifications` before. I don't think it is necessary, especially because that is implementation detail. Either the user cares about logging and want to filter to some particular messages, or they want all of them.